scene/2d/skeleton_2d.cpp

```diff
@@ -429,29 +429,22 @@ PackedStringArray Bone2D::get_configuration_warnings() const {
 }
 
 void Bone2D::calculate_length_and_rotation() {
-	// if there is at least a single child Bone2D node, we can calculate
+	// If there is at least a single child Bone2D node, we can calculate
 	// the length and direction. We will always just use the first Bone2D for this.
-	bool calculated = false;
 	int child_count = get_child_count();
-	if (child_count > 0) {
-		for (int i = 0; i < child_count; i++) {
-			Bone2D *child = Object::cast_to<Bone2D>(get_child(i));
-			if (child) {
-				Vector2 child_local_pos = to_local(child->get_global_position());
-				length = child_local_pos.length();
-				bone_angle = child_local_pos.normalized().angle();
-				calculated = true;
-				break;
-			}
+
+	for (int i = 0; i < child_count; i++) {
+		Bone2D *child = Object::cast_to<Bone2D>(get_child(i));
+		if (child) {
+			Vector2 child_local_pos = to_local(child->get_global_position());
+			length = child_local_pos.length();
+			bone_angle = child_local_pos.normalized().angle();
```

Comment:
Did you mean `global_inv.xform(child->get_global_position())`?  

Is okay doing that? I don't know nothing about `ERR_READ_THREAD_GUARD_V(Point2());` that is in `Node2D::to_local`.

Reply:
> Did you mean `global_inv.xform(child->get_global_position())`?

Oh! Yes, indeed. :+1:

> Is okay doing that? I don't know nothing about `ERR_READ_THREAD_GUARD_V(Point2());` that is in `Node2D::to_local`.

It's also in `CanvasItem::get_global_transform` and `Node2D::get_global_position`, and in case these guards would fail the resulting `child_local_pos` will end up being the same (result of `Transform2D().xform(Vector2())` so `Vector2()`). So yes, this seems okay / shouldn't change the behavior anyhow.

Reply:
Make sense.

Updated!